native/src/image.rs

```diff
@@ -84,14 +85,70 @@ impl Hash for Handle {
     }
 }
 
+/// Reference counted wrapper around raw image data.
+///
+/// Behaves like a `&[u8]`.
+pub struct ImageBytes(Box<dyn ImageBytesTrait>);
+
+trait ImageBytesTrait: AsRef<[u8]> + Send + Sync + 'static {
+    fn clone_boxed(&self) -> Box<dyn ImageBytesTrait>;
```

Comment:
The `Clone` trait bound everywhere and this new trait wouldn't be needed if it would be OK to not have `Data` implement `Clone`.

With `Handle` being `Clone`, I don't see much use for it but I might be missing something. What do you think?